src/libp2p/discovery.rs

```diff
@@ -429,6 +429,10 @@ impl NetworkBehaviour for DiscoveryBehaviour {
                                     Some(info.clone());
                                 if let Some(kademlia) = self.discovery.kademlia.as_mut() {
                                     for address in &info.listen_addrs {
+                                        if self.n_node_connected >= self.target_peer_count {
```

Comment:
This should not be inside the for loop

Reply:
Thanks for your contribution! However, with the current Kademlia config, addresses are automatically inserted into the routing table (see https://docs.rs/libp2p-kad/0.46.2/libp2p_kad/enum.BucketInserts.html). So I doubt if the change in this PR makes any difference. It'd be great to have unit tests for this scenario (we have not got a chance to prioritize the work yet)

> Whenever a connection to a peer is established as a result of a dialing attempt and that peer is not yet in the routing table, it is inserted as long as there is a free slot in the corresponding k-bucket. If the k-bucket is full but still has a free pending slot, it may be inserted into the routing table at a later time if an unresponsive disconnected peer is evicted from the bucket.